cmd/juju/application/deployer/charm.go

```diff
@@ -389,9 +390,9 @@ func (c *repositoryCharm) PrepareAndDeploy(ctx *cmd.Context, deployAPI DeployerA
 			Trust:            c.trust,
 		})
 
-		uploadErr := c.uploadExistingPendingResources(c.applicationName, localPendingResources, deployAPI, c.model.Filesystem())
+		uploadErr := c.uploadExistingPendingResources(appName, localPendingResources, deployAPI, c.model.Filesystem())
```

Comment:
Should be info.Name instead. The application name is user provided or taken from the metadata.yaml file. Therefore the charm name and application name are not guaranteed to be the same independent of the user providing an application name.

Side note, there is a bug in the server side as the name in metadata.yaml isn't always used.  The `jameinel-ubuntu-lite` charm should deploy as an `ubuntu-lite` application.